packages/common/src/hardhat/tasks/collateralWhitelist.ts

```diff
@@ -0,0 +1,238 @@
+        findL2TokenForL1Token(l2Web3, l2ChainId, l1TokenWhitelist.l1TokenAddress)
+      )
+    );
+
+    // Remove any tokens that are not found on L2.
+    const combineSet = l1TokenWhitelistArray
+      .map((l1TokenWhitelist: any, index: any) => {
+        return { ...l1TokenWhitelist, l2TokenAddress: associatedL2Tokens[index] };
+      })
+      .filter((tokenList: any) => tokenList.l2TokenAddress !== ZERO_ADDRESS);
+
+    console.log(`Found the following ${combineSet.length} L1->L2 mapping and the associated final fees`);
+    console.table(combineSet);
+
+    console.log("Removing any tokens that are already on the L2 whitelist...");
+    const l2TokenWhitelistContract = await deployments.get("AddressWhitelist");
+    const l2TokenWhitelist = new l2Web3.eth.Contract(l2TokenWhitelistContract.abi, l2TokenWhitelistContract.address);
+    const currentBlock = await l2Web3.eth.getBlockNumber();
+    const blockLookBack = 99999; // We need to use paginated query on L2 as some L2s limit how far you can look back
+    // such as arbitrum which has a 100k block lookback restriction.
+
+    const eventResults = await Promise.all([
+      getEventsWithPaginatedBlockSearch(
+        l2TokenWhitelist as Web3Contract,
+        "AddedToWhitelist",
+        0,
+        currentBlock,
+        blockLookBack
+      ),
+      getEventsWithPaginatedBlockSearch(
+        l2TokenWhitelist as Web3Contract,
+        "RemovedFromWhitelist",
+        0,
+        currentBlock,
+        blockLookBack
+      ),
+    ]);
+
+    const l2AddedToWhitelistTokens = eventResults[0].eventData.map((event) => event.returnValues.addedAddress);
+    const l2RemovedFromWhitelistTokens = eventResults[1].eventData.map((event) => event.returnValues.removedAddress);
+
+    const l2WhitelistedAddressArray = l2AddedToWhitelistTokens.filter((address) => {
+      return !l2RemovedFromWhitelistTokens.includes(address);
+    });
+
+    const filteredCombinedSet = combineSet.filter((element: any) => {
+      return !l2WhitelistedAddressArray.includes(element.l2TokenAddress);
+    });
+
+    console.log(
+      `Adding ${filteredCombinedSet.length} tokens the the L2 token whitelist on ${l2TokenWhitelist.options.address}...`
+    );
+    console.table(filteredCombinedSet);
+    const l2StoreContract = await deployments.get("AddressWhitelist");
+    const l2Store = new l2Web3.eth.Contract(l2StoreContract.abi, l2StoreContract.address);
+
+    let nonce = await l2Web3.eth.getTransactionCount(deployer);
+    for (let index = 0; index < filteredCombinedSet.length; index++) {
+      console.log(
+        `Whitelisting ${filteredCombinedSet[index].symbol} at ${filteredCombinedSet[index].l1TokenAddress} with fee ${filteredCombinedSet[index].finalFee}...`
+      );
+      const whitelistTx = await l2TokenWhitelist.methods
+        .addToWhitelist(filteredCombinedSet[index].l2TokenAddress)
+        .send({ from: deployer, nonce });
+      nonce++;
```

Comment:
Why are you setting nonce? Can't you just use `.send({ from: })`?

Reply:
for example I will often get this:
![image](https://user-images.githubusercontent.com/12886084/148774965-8bc20acf-ab23-4020-b84c-ba6e43fd8f48.png)